src/graphql/mod.rs

```diff
@@ -97,21 +98,71 @@ impl QueryClient {
         Q::ResponseData: Debug,
     {
         let body = Q::build_query(variables);
-        let request = self.inner().post(&self.api_url).json(&body);
 
         debug!("url: {:?}", &self.api_url);
-        debug!("request: {:#?}", request);
         debug!("GraphQL query: \n{}", body.query);
 
-        let response: Response<Q::ResponseData> = request.send().await?.json().await?;
+        let response: Result<Response<Q::ResponseData>, APIError> =
+            self.retry_request::<Q>(body, handler).await;
         debug!("GraphQL response: {:#?}", response);
 
-        if let Some(errors) = response.errors.clone() {
-            let first_error = errors[0].clone();
+        response
+    }
 
-            match check_auth_error(&first_error) {
-                Some(err) => return Err(err),
-                None => return handler(response, first_error),
+    // Attempts the request and retries up to 3 times if the request times out.
+    async fn retry_request<Q>(
+        &self,
+        body: QueryBody<Q::Variables>,
+        handler: impl Fn(
+            Response<Q::ResponseData>,
+            graphql_client::Error,
+        ) -> Result<Response<Q::ResponseData>, APIError>,
+    ) -> Result<Response<Q::ResponseData>, APIError>
+    where
+        Q: GraphQLQuery,
+        Q::ResponseData: Debug,
+    {
+        let mut retry_count = 0;
+        let request = self.inner().post(&self.api_url).json(&body);
+
+        debug!("GraphQL equest: {:#?}", request);
```

Comment:
typo here

Reply:
and here say `GraphQL request`, but below say `request`, I think we should make them consistent